sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java

```diff
@@ -284,6 +285,15 @@ private SpecializedBlobClientBuilder prepareBuilder() {
      * @return A reactive response containing the information of the uploaded block blob.
      */
     public Mono<Response<BlockBlobItem>> uploadWithResponse(Flux<ByteBuffer> data,
+        ParallelTransferOptions parallelTransferOptions, BlobHttpHeaders headers, Map<String, String> metadata,
+        AccessTier tier, BlobRequestConditions accessConditions) {
+        return determineUploadFullOrChunked(data, (stream) ->
+                uploadInChunks(stream, parallelTransferOptions, headers, metadata, tier, accessConditions),
+            (stream, length) -> getBlockBlobAsyncClient()
+                .uploadWithResponse(stream, length, headers, metadata, tier, accessConditions));
```

Comment:
We should pass ProgressReporter.addProgressReporting(stream) into upload()

Reply:
Done